common/djangoapps/student/views.py

```diff
@@ -254,6 +254,19 @@ def get_course_enrollments(user, org_to_include, orgs_to_exclude):
     for enrollment in CourseEnrollment.enrollments_for_user(user):
 
         # If the course is missing or broken, log an error and skip it.
+        #
+        # TODO (RFL): Enrollment now creates a CourseOverview; enrollments
+        # in deleted courses will have a stale CourseOverview. As a result,
+        # this method no longer ignores course enrollments for deleted courses.
+        # Course deletion needs to bust the CourseOverview cache; the
+        # CourseOverview stored on enrollments will become stale after this,
+        # and it will be necessary to make an additional SQL query instead of
+        # relying on the _course_overview variable scoped to the enrollment.
+        #
+        # How does this method deal with course deletion now? Accessing an
+        # enrollment's course_overview property creates a CourseOverview.
+        # If the cached course is then deleted, any enrollments in it will
+        # continue to be listed (since it has a valid CourseOverview).
```

Comment:
@mekkz and I discussed this offline. Although course publication emits a signal which busts the CourseOverview cache, course deletion does not. This issue only became apparent as part of this PR because CourseOverviews are now created during enrollment &mdash; earlier than in the past &mdash; causing tests which expect enrollments in deleted courses to be excluded from the student's course listing to fail.

A proposed solution is to emit a signal from the modulestore's course deletion method whose receiver busts the CourseOverview cache for the deleted course. I'll give this a shot, but would appreciate eyes from someone on the platform team (e.g., @doctoryes).